### PR TITLE
DOC: Completed "Examples" sections of all linalg funcs

### PR DESCRIPTION
**Pull request by ilayn**
MAINT: Touched `lu_factor` -> LinAlgWarning and stacklevel
MAINT: Touched `qr_multiply` -> Cleaned up the code and fixed an ndim bug
MAINT: Touched `rsf2csf` -> Cleaned up code

It is mostly documentation additions so in case docstring test passes the review load is minor. The parts that need attention are items above. 

Cross-ref : #7168

### CODE REVIEW

**Comment by ilayn** [comment]
Does anyone know what

    ValueError: >>> from scipy.linalg import eigvals is not a item name

means in CircleCI error? I can't replicate it locally.

**Comment by person142** [comment]
First guess would be it's the lack of a blank line between the end of the `See Also` section and the `Examples` section for `eigvals`.

**Comment by ilayn** [comment]
Argh.... *facepalm*

**Comment by ilayn** [comment]
All green and I think this is ready for review.

File: scipy/linalg/_procrustes.py [hunk 1]
```diff
@@ -16,10 +16,7 @@ def orthogonal_procrustes(A, B, check_finite=True):
     Compute the matrix solution of the orthogonal Procrustes problem.
 
     Given matrices A and B of equal shape, find an orthogonal matrix R
-    that most closely maps A to B [1]_.
-    Note that unlike higher level Procrustes analyses of spatial data,
-    this function only uses orthogonal transformations like rotations
-    and reflections, and it does not use scaling or translation.
+    that most closely maps A to B using the algorithm given in [1]_.
```
Comment (ev-br) [inline scipy/linalg/_procrustes.py]: FWIW, I think the original comment was useful.

Comment (ev-br) [reply scipy/linalg/_procrustes.py]: Scratch that, I see now that the comment is moved to the Notes section.

**Comment by ilayn** [comment]
Link for the rendered versions of the docstrings  https://5060-1460385-gh.circle-artifacts.com/0/home/ubuntu/scipy/doc/build/html-scipyorg/linalg.html

File: scipy/linalg/_decomp_qz.py [hunk 2]
```diff
@@ -342,6 +341,16 @@ def ordqz(A, B, sort='lhp', output='real', overwrite_a=False,
     --------
     qz
 
+    Examples
+    --------
+    >>> from scipy.linalg import ordqz
+    >>> A = np.array([[2, 5, 8, 7], [5, 2, 2, 8], [7, 5, 6, 6], [5, 4, 4, 8]])
+    >>> B = np.array([[0, 6, 0, 0], [5, 0, 2, 1], [5, 2, 6, 6], [4, 7, 7, 7]])
+    >>> AA, BB, alpha, beta, Q, Z = ordqz(A, B, sort='lhp')
+    >>> # since we have sorted for left half plane eig, negatives come first
```
Comment (rgommers) [inline scipy/linalg/_decomp_qz.py]: Style nitpick: *blank line - sentence - blank line* would look better than as a comment

File: scipy/linalg/_procrustes.py [hunk 3]
```diff
@@ -36,28 +33,43 @@ def orthogonal_procrustes(A, B, check_finite=True):
...
     References
     ----------
     .. [1] Peter H. Schonemann, "A generalized solution of the orthogonal
            Procrustes problem", Psychometrica -- Vol. 31, No. 1, March, 1996.
 
+    Examples
+    --------
+    >>> from scipy.linalg import orthogonal_procrustes
+    >>> A = np.array([[ 2,  0,  1], [-2,  0,  0]])
+    >>> # Flip the order of cols and check for the anti-diagonal mapping
```
Comment (rgommers) [inline scipy/linalg/_procrustes.py]: same here, sentence > comment

File: scipy/linalg/decomp.py [hunk 4]
```diff
@@ -910,6 +978,17 @@ def eigvalsh_tridiagonal(d, e, select='a', select_range=None,
     --------
     eigh_tridiagonal : eigenvalues and right eiegenvectors for
         symmetric/Hermitian tridiagonal matrices
+
+    Examples
+    --------
+    >>> from scipy.linalg import eigvalsh_tridiagonal, eigvalsh
+    >>> from numpy import allclose, zeros, ones, diag
```
Comment (rgommers) [inline scipy/linalg/decomp.py]: better to remove these imports and just use ``np.``

File: scipy/linalg/decomp.py [hunk 5]
```diff
@@ -910,6 +978,17 @@ def eigvalsh_tridiagonal(d, e, select='a', select_range=None,
...
+    >>> d, e = 3*ones(4), -1*ones(3)
```
Comment (rgommers) [inline scipy/linalg/decomp.py]: clearer to put on two separate lines

File: scipy/linalg/decomp.py [hunk 6]
```diff
@@ -1109,6 +1198,19 @@ def hessenberg(a, calc_q=False, overwrite_a=False, check_finite=True):
         Unitary/orthogonal similarity transformation matrix ``A = Q H Q^H``.
         Only returned if ``calc_q=True``.
 
+    Examples
+    --------
+    >>> from scipy.linalg import hessenberg
+    >>> from numpy import allclose, zeros
```
Comment (rgommers) [inline scipy/linalg/decomp.py]: use ``np.`` instead

File: scipy/linalg/decomp_cholesky.py [hunk 7]
```diff
@@ -137,6 +137,20 @@ def cho_factor(a, lower=False, overwrite_a=False, check_finite=True):
     cho_solve : Solve a linear set equations using the Cholesky factorization
                 of a matrix.
 
+    Examples
+    --------
+    >>> from scipy.linalg import cho_factor
+    >>> from numpy import allclose, zeros, triu
```
Comment (rgommers) [inline scipy/linalg/decomp_cholesky.py]: use ``np.`` instead

Comment (rgommers) [reply scipy/linalg/decomp_cholesky.py]: same in more places, won't comment on it again

File: scipy/linalg/decomp_lu.py [hunk 8]
```diff
@@ -53,6 +53,17 @@ def lu_factor(a, overwrite_a=False, check_finite=True):
     -----
     This is a wrapper to the ``*GETRF`` routines from LAPACK.
 
+    Examples
+    --------
+    >>> from scipy.linalg import lu_factor
+    >>> from numpy import tril, triu, allclose, zeros, eye
+    >>> A = np.array([[2, 5, 8, 7], [5, 2, 2, 8], [7, 5, 6, 6], [5, 4, 4, 8]])
+    >>> lu, piv = lu_factor(A)
+    >>> # piv is [2, 2, 3, 3]
```
Comment (rgommers) [inline scipy/linalg/decomp_lu.py]: evaluate `piv` instead?

File: scipy/linalg/decomp_schur.py [hunk 9]
```diff
@@ -190,58 +217,75 @@ def rsf2csf(T, Z, check_finite=True):
...
-            T[k,j] = dot(G, T[k,j])
-            i = slice(0, m+1)
-            T[i,k] = dot(T[i,k], Gc)
-            i = slice(0, N)
-            Z[i,k] = dot(Z[i,k], Gc)
-        T[m,m-1] = 0.0
+        if abs(T[m, m-1]) > eps*(abs(T[m-1, m-1]) + abs(T[m, m])):
+            mu = eigvals(T[m-1:m+1, m-1:m+1]) - T[m, m]
+            r = norm([mu[0], T[m, m-1]])
+            c, s = mu[0] / r, T[m, m-1] / r
```
Comment (rgommers) [inline scipy/linalg/decomp_schur.py]: better to keep this on separate lines

**Review by rgommers — CHANGES_REQUESTED** [review]
Overall LGTM, missing test for the bugfix is my only nontrivial comment.

For the record: ``@`` is not yet valid syntax for all supported Python versions, but I approve of it in these examples:)

**Comment by ilayn** [comment]
Rebased and squashed

**Review by rgommers — APPROVED** [review]

**Comment by rgommers** [comment]
LGTM now, merged. Thanks @ilayn!

**Pull request merged**